resources/[gameplay]/chat/cl_chat.lua

```diff
@@ -306,3 +307,25 @@ Citizen.CreateThread(function()
     end
   end
 end)
+
+local function setHideState(state,setLastState)
+  if type(state) == 'string' then
+    state = CHAT_HIDE_STATES[state]
+  end
+
+  chatHideState = state
+
+  if setLastState then
+    lastChatHideState = state
+  end
+end

```

Comment:
```suggestion
  SetResourceKvp('hideState', **tostring(chatHideState))**
end
```
Save changes to KVP.